src/analysis/Environments/Environments.models.ts

```diff
@@ -17,9 +17,11 @@ export type AppWithWorkspace = DecoratedResourceAttributes & ListAppItem;
 export type DecoratedComputeResource = RuntimeWithWorkspace | AppWithWorkspace;
 export type DecoratedResource = DecoratedComputeResource | DiskWithWorkspace;
 
-export interface LeoResourcePermissionsProvider {
+export interface PermissionsAndStateProvider {
   canDeleteDisk: (disk: PersistentDisk) => boolean;
   canPauseResource: (resource: App | ListRuntimeItem) => boolean;
+  canDeleteApp: (resource: DecoratedComputeResource) => boolean;
```

Comment:
can the type of this not be Decorated variant?  how about App | PersistentDisk | Runtime like canDeleteResource below?

Reply:
Keeping that logic in mind, I think it makes even more sense to have this type be `App` since we're not passing in any other types. What do you think?

Reply:
I think it makes sense for this to take an `App`. We don't care that it is decorated here.